core/src/bundle_stage.rs

```diff
@@ -122,6 +123,53 @@ struct AllExecutionResults {
     pub post_balances: (TransactionBalances, TransactionTokenBalances),
 }
 
+struct BundleReservedSpace {
+    current_tx_block_limit: u64,
+    current_bundle_block_limit: u64,
+    initial_allocated_cost: u64,
+}
+
+impl BundleReservedSpace {
+    fn reset_reserved_cost(&mut self, working_bank: &Arc<Bank>) {
+        self.current_bundle_block_limit = MAX_BLOCK_UNITS;
+        self.current_tx_block_limit = MAX_BLOCK_UNITS - self.initial_allocated_cost;
```

Comment:
use saturating math here and elsewhere. master doesn't build right now but when it does it'll compain about that